docs/reference/specifications/providers.md

```diff
@@ -0,0 +1,251 @@
+| ------------------ | -------------------------- | ------------------------------------------------------------------------------- | ------------------------ | ----------------------------- | ------------------- |
+| resolver           | FLAGD_RESOLVER             | mode of operation                                                               | String - rpc, in-process | rpc                           | rpc & in-process    |
+| host               | FLAGD_HOST                 | remote host                                                                     | String                   | localhost                     | rpc & in-process    |
+| port               | FLAGD_PORT                 | remote port                                                                     | int                      | 8013 (rpc), 8015 (in-process) | rpc & in-process    |
+| targetUri          | FLAGD_GRPC_TARGET          | alternative to host/port, supporting custom name resolution                     | string                   | null                          | rpc & in-process    |
+| tls                | FLAGD_TLS                  | connection encryption                                                           | boolean                  | false                         | rpc & in-process    |
+| socketPath         | FLAGD_SOCKET_PATH          | alternative to host port, unix socket                                           | String                   | null                          | rpc & in-process    |
+| certPath           | FLAGD_SERVER_CERT_PATH     | tls cert path                                                                   | String                   | null                          | rpc & in-process    |
+| deadline           | FLAGD_DEADLINE_MS          | deadline for unary calls, and timeout for initialization                        | int                      | 500                           | rpc & in-process    |
+| streamDeadlineMs   | FLAGD_STREAM_DEADLINE_MS   | deadline for streaming calls, useful as an application-layer keepalive          | int                      | 600000                        | rpc & in-process    |
+| retryBackoffMs     | FLAGD_RETRY_BACKOFF_MS     | initial backoff for stream retry                                                | int                      | 1000                          | rpc & in-process    |
+| retryBackoffMaxMs  | FLAGD_RETRY_BACKOFF_MAX_MS | maximum backoff for stream retry                                                | int                      | 120000                        | rpc & in-process    |
+| retryGraceAttempts | FLAGD_RETRY_GRACE_ATTEMPTS | amount of stream retry attempts before provider moves from STALE to ERROR state | int                      | 5                             | rpc & in-process    |
```

Comment:
Ya, 15-30 seemed correct to me, personally, since I think most services and infra would be able to cycle within that time. 5 was more or less based on that idea.

Open to other arguments here though!